src/server/system_services/system_store.js

```diff
@@ -804,6 +804,12 @@ class SystemStore extends EventEmitter {
         }
     }
 
+    check_for_access_key_duplication(access_key_id) {
+        return _.some(this.data.accounts, acc =>
+            _.some(acc.access_keys, key => key.access_key.unwrap() === access_key_id.unwrap())
+        );
+    }
```

Comment:
I think a function like `check_for_access_key_duplication` is too specific to implement in the system_store. 
you can change it to `get_account_by_access_key` that is more generic, or use `find` directly in `account_server` without adding a function.

```suggestion
    get_account_by_access_key(access_key_id) {
        return _.find(this.data.accounts, acc =>
        acc.access_keys && acc.access_keys[0].access_key.unwrap() === access_key_id.unwrap()
    );
    }
```

Reply:
@dannyzaken For more general code we can iterate all the access keys in the array of `access_keys` and not just the first element (although today we replace the whole array with an array with one `access_keys`). WDYT?